### PR TITLE
Updates CapabilityData to use AlgorithmPropertyList

### PR DESCRIPTION
**Pull request by Superhepper**
- Adds the types AlgorithmAttributes, AlgorithmProperty and AlgorithmPropertyList and their associated tests.
- Changes the type in CapabilityData::Algorithms to AlgorithmPropertyList .

### CODE REVIEW

**Review by ionut-arm — COMMENTED** [review]
🙏🏻 thanks for the patch!

File: tss-esapi/src/attributes/algorithm.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    pub encrypting, _: 9;
+    pub method, _: 10;
+    // 31:11 Reserved
+}
+
+impl From<TPMA_ALGORITHM> for AlgorithmAttributes {
+    fn from(tpma_algorithm: TPMA_ALGORITHM) -> Self {
+        AlgorithmAttributes(tpma_algorithm)
+    }
+}
```
[the remark this answers is not in the document]
Comment (Superhepper) [reply tss-esapi/src/attributes/algorithm.rs:28]: I only do that when the specification explicitly says they should not be set.

File: tss-esapi/src/structures/algorithm_property.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/// algorithm.
+///
+/// # Details
+/// This corresponds to the TPMS_ALG_PROPERTY
+/// structure.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct AlgorithmProperty {
+    algorithm_identifier: AlgorithmIdentifier,
+    algorithm_properties: AlgorithmAttributes,
+}
```
[the remark this answers is not in the document]
Comment (Superhepper) [reply tss-esapi/src/structures/algorithm_property.rs:20]: Well you can sort of already do that with ```find``` method and then compare the properties.

File: tss-esapi/tests/integration_tests/attributes_tests/algorithm_attributes_tests.rs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+#[test]
+fn test_method_set() {
+    let attributes = AlgorithmAttributes::from(1u32.shl(10));
+    assert!(!attributes.asymmetric(), "'asymmetric' is unexpectedly set");
+    assert!(!attributes.symmetric(), "'symmetric' is unexpectedly set");
+    assert!(!attributes.hash(), "'hash' is unexpectedly set");
+    assert!(!attributes.object(), "'object' is unexpectedly set");
+    assert!(!attributes.signing(), "'signing' is unexpectedly set");
+    assert!(!attributes.encrypting(), "'encrypting' is unexpectedly set");
+    assert!(attributes.method(), "'method' is unexpectedly not set");
```
[the remark this answers is not in the document]
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/attributes_tests/algorithm_attributes_tests.rs:149]: Well I just want to be sure that I have not specified the wrong bits when specifying the bitfield.

File: tss-esapi/tests/integration_tests/common/tpms_types_equality_checks.rs [hunk 4]
```diff
@@ -223,3 +224,15 @@ pub fn ensure_tpms_tagged_property_equality(
         "'value' value in TPMS_TAGGED_PROPERTY, mismatch between actual and expected",
     );
 }
+
+#[allow(dead_code)]
```
Comment (wiktor-k) [inline tss-esapi/tests/integration_tests/common/tpms_types_equality_checks.rs:228]: Why is the `#[allow(dead_code)]` present here? The code looks straightforward... 🤔

File: tss-esapi/tests/integration_tests/structures_tests/lists_tests/algorithm_property_list_tests.rs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    let expected_algorithm_property_list: AlgorithmPropertyList = expected_tpml_alg_property
+        .try_into()
+        .expect("Failed to convert TPML_ALG_PROPERTY into AlgorithmPropertyList");
+
+    assert_eq!(
+        expected_algorithm_properties.len(), expected_algorithm_property_list.len(),
+        "The algorithm property list converted from TPML_ALG_PROPERTY did not contain the expected amount of elements"
+    );
+
+    let algorithm_properties_vector: Vec<AlgorithmProperty> =
```
Comment (wiktor-k) [inline tss-esapi/tests/integration_tests/structures_tests/lists_tests/algorithm_property_list_tests.rs:88]: Usually `Vec<_>` is sufficient for type inference...

**Review by wiktor-k — COMMENTED** [review]
This looks nice and clean 👍

File: tss-esapi/tests/integration_tests/common/tpms_types_equality_checks.rs (hunk 4 above)
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/common/tpms_types_equality_checks.rs:228]: I had some issues with the compiler not understand that the functions were in use in the tests.

File: tss-esapi/tests/integration_tests/structures_tests/lists_tests/algorithm_property_list_tests.rs (hunk 5 above)
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/structures_tests/lists_tests/algorithm_property_list_tests.rs:88]: Nice, to know. I like being explicit though unless I am feeling lazy.

File: tss-esapi/src/structures/algorithm_property.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Superhepper) [reply tss-esapi/src/structures/algorithm_property.rs:20]: Fixed, added a 'new' associated function.

File: tss-esapi/tests/integration_tests/common/tpms_types_equality_checks.rs (hunk 4 above)
Comment (Superhepper) [reply tss-esapi/tests/integration_tests/common/tpms_types_equality_checks.rs]: Fixed

File: tss-esapi/src/structures/algorithm_property.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ionut-arm) [reply tss-esapi/src/structures/algorithm_property.rs:20]: Thanks!

**Review by ionut-arm — APPROVED** [review]
👍🏻

**Pull request merged**